libs/application/core/src/validation/validators.ts

```diff
@@ -92,10 +94,20 @@ function partialSchemaValidation(
           try {
             trimmedSchema.parse({ [key]: [el] })
           } catch (e) {
+            let schemaShape = trimmedSchema?.shape[key]?._def?.type
```

Comment:
Feel like some tests would be in order for this before we go any further showing that this solves the issue mentioned in the why.

Reply:
Hehe,
Good point, on it :)